packages/@jsii/java-runtime/project/src/main/java/software/amazon/jsii/JsiiEngine.java

```diff
@@ -486,7 +486,11 @@ private Object invokeMethod(final Object obj, final Method method, final Object.
                 throw e;
             }
         } catch (InvocationTargetException e) {
-            throw new JsiiError(e.getTargetException());
+            if (e.getTargetException().getClass().equals(JsiiError.class)) {
+                throw new JsiiError(e.getTargetException());
+            } else {
+                throw new RuntimeException(e.getTargetException());
+            }
```

Comment:
Similarly to above, I would:

```suggestion
            } else if (e.getTargetException() instanceof RuntimeException) {
                throw (RuntimeException)(e.getTargetException();
            } else {
                // Can't just throw a checked error without wrapping it :(
                throw new RuntimeException(e.getTargetException());
            }
```